lib/environment.js

```diff
@@ -125,3 +134,34 @@ Environment.prototype.updateConfigVars = function (pkgConfigs) {
 
   return this.env;
 };
+
+/**
+ * Update environment `--env` / `--env-path` command line options.
+ *
+ * @param   {Object}  opts          Options
+ * @param   {String}  opts.env      Stringified JSON object
+ * @param   {String}  opts.envPath  Path to JSON file
+ * @returns {Object}                Mutated environment variable.
+ */
+Environment.prototype.updateEnvFlag = function (opts) {
+  // Nothing to parse.
+  if (!(opts.env || opts.envPath)) {
+    return this.env;
+  }
+
+  // Parse envs.
+  var envsObj = jsonParse({
+    str: opts.env,
+    path: opts.envPath
+  });
+
+  // Validation
+  if (_.isEmpty(envsObj)) {
+    throw new Error("Empty/null JSON environment.");
```

Comment:
I could see this being perfectly intentional/allowable behavior, especially if people are running `builder` as part of a larger setup where their scripts just always pass `--env` / `--env-path` but they sometimes contain no overrides.